js/src/dropdown.js

```diff
@@ -357,6 +357,33 @@ class Dropdown extends BaseComponent {
     }
   }
 
+  _selectMenuItem(event) {
+    const items = SelectorEngine.find(SELECTOR_VISIBLE_ITEMS, this._menu).filter(isVisible)
+
+    if (!items.length) {
+      return
+    }
+
+    let index = items.indexOf(event.target)
+
+    // Up
+    if (event.key === ARROW_UP_KEY && index > 0) {
+      index--
+    }
+
+    // Down
+    if (event.key === ARROW_DOWN_KEY && index < items.length - 1) {
+      index++
+    }
+
+    // index is -1 if the first keydown is an ArrowUp
+    if (index < 0) {
```

Comment:
The comment is redundant now :slightly_smiling_face:
And I am also not sure why you updated this check :thinking:

Reply:
I was to implement #32954 as well and the code been like

```js
// Up
if (event.key === ARROW_UP_KEY) {
  index--
}

// Down
if (event.key === ARROW_DOWN_KEY) {
  index++
}

if (index < 0) {
  index = items.length - 1
} else if (index === items.length) {
  index = 0
}
```

But then I changed it back because I wasn't sure if I should do this in scope of that PR. 🙂 Sorry forgot about that block.

The comment is misleading anyway because the -1 is inferred by `indexOf()` with `event.target` being the button when there is no menu item with focus. The other conditions will check for `index > 0` and `index < items.length - 1`. So it's rather `index is -1 for the first keydown`.

Reply:
Reverted the condition :+1: